numba_cuda/numba/cuda/dispatcher.py

```diff
@@ -237,16 +236,18 @@ def maybe_link_nrt(self, link, tgt_ctx, asm):
         if not tgt_ctx.enable_nrt:
             return
 
-        all_nrt = "|".join(self.NRT_functions)
-        pattern = (
-            r'\.extern\s+\.func\s+(?:\s*\(.+\)\s*)?('
-            + all_nrt + r')\s*\([^)]*\)\s*;'
-        )
-
-        nrt_in_asm = re.findall(pattern, asm)
-
-        basedir = os.path.dirname(os.path.abspath(__file__))
-        if nrt_in_asm:
+        nrt_in_asm = lambda asm: any(fn in asm for fn in self.NRT_functions)
+        link_nrt = nrt_in_asm(asm)
+        if not link_nrt:
+            for file in link:
+                if file.endswith('ptx'):
```

Comment:
Alternatively, there can also be `linkable_code.PTXSource` in the input. Could there be input in other format that also contains NRT calls?

Reply:
Thats a reasonable question. This would only currently catch the case where the user supplied a PTX file that utilized NRT functions. I suppose it could also occur with a `.cu` file, which it shouldn't be hard to update things for.

Reply:
https://github.com/NVIDIA/numba-cuda/pull/167/commits/ff478f462ebf186f59a0104db9762ad396cc276f